src/library_browser.js

```diff
@@ -1475,13 +1532,17 @@ var LibraryBrowser = {
 
   emscripten_call_worker__proxy: 'sync',
   emscripten_call_worker__sig: 'viiiiii',
+#if !MINIMAL_RUNTIME
+  emscripten_call_worker__deps: ['$runtimeKeepalivePush'],
+#endif
   emscripten_call_worker: function(id, funcName, data, size, callback, arg) {
-    noExitRuntime = true; // should we only do this if there is a callback?
-
     funcName = UTF8ToString(funcName);
     var info = Browser.workers[id];
     var callbackId = -1;
     if (callback) {
+#if !MINIMAL_RUNTIME
+      {{{ runtimeKeepalivePush() }}}
```

Comment:
it's not easy to see what pop corresponds to this (it's earlier up). how about a comment?